prog/dftb+/lib_dftb/periodic.F90

```diff
@@ -713,19 +715,67 @@ subroutine reallocateArrays3(iNeighbor, neighDist2, mNewNeighbor)
 
   end subroutine reallocateArrays3
 
+  !> Calculate indexing array and number of elements in sparse arrays like the real space overlap
+  subroutine getSparseDescriptor(iNeighbor, nNeighbor, img2CentCell, orb, iPair, sparseSize)
+
+    !> Neighbours of each atom
+    integer, intent(in) :: iNeighbor(0:,:)
+
+    !> Number of neighbours of each atom
+    integer, intent(in) :: nNeighbor(:)
+
+    !> Indexing for mapping image atoms to central cell
+    integer, intent(in) :: img2CentCell(:)
+
+    !> Atomic orbital information
+    type(TOrbitals), intent(in) :: orb
+
+    !> Sparse array indexing for the start of atomic blocks in data structures
+    integer, allocatable, intent(inout) :: iPair(:,:)
```

Comment:
Must be intent(inout), as an intent(out) allocatable argument is automatically deallocated when entering the subroutine.